libflux/src/ast/walk/mod.rs

```diff
@@ -0,0 +1,363 @@
+            Node::ConditionalExpr(_) => write!(f, "ConditionalExpr"),
+            Node::StringExpr(_) => write!(f, "StringExpr"),
+            Node::ParenExpr(_) => write!(f, "ParenExpr"),
+            Node::IntegerLit(_) => write!(f, "IntegerLit"),
+            Node::FloatLit(_) => write!(f, "FloatLit"),
+            Node::StringLit(_) => write!(f, "StringLit"),
+            Node::DurationLit(_) => write!(f, "DurationLit"),
+            Node::UintLit(_) => write!(f, "UintLit"),
+            Node::BooleanLit(_) => write!(f, "BooleanLit"),
+            Node::DateTimeLit(_) => write!(f, "DateTimeLit"),
+            Node::RegexpLit(_) => write!(f, "RegexpLit"),
+            Node::PipeLit(_) => write!(f, "PipeLit"),
+            Node::BadExpr(_) => write!(f, "BadExpr"),
+            Node::ExprStmt(_) => write!(f, "ExprStmt"),
+            Node::OptionStmt(_) => write!(f, "OptionStmt"),
+            Node::ReturnStmt(_) => write!(f, "ReturnStmt"),
+            Node::BadStmt(_) => write!(f, "BadStmt"),
+            Node::TestStmt(_) => write!(f, "TestStmt"),
+            Node::BuiltinStmt(_) => write!(f, "BuiltinStmt"),
+            Node::Block(_) => write!(f, "Block"),
+            Node::Property(_) => write!(f, "Property"),
+            Node::TextPart(_) => write!(f, "TextPart"),
+            Node::InterpolatedPart(_) => write!(f, "InterpolatedPart"),
+            Node::VariableAssgn(_) => write!(f, "VariableAssgn"),
+            Node::MemberAssgn(_) => write!(f, "MemberAssgn"),
+        }
+    }
+}
+
+impl<'a> Node<'a> {
+    fn from_expr(expr: &'a Expression) -> Node {
+        match expr {
+            Expression::Identifier(e) => Node::Identifier(e),
+            Expression::Array(e) => Node::ArrayExpr(e),
+            Expression::Function(e) => Node::FunctionExpr(e),
+            Expression::Logical(e) => Node::LogicalExpr(e),
+            Expression::Object(e) => Node::ObjectExpr(e),
+            Expression::Member(e) => Node::MemberExpr(e),
+            Expression::Index(e) => Node::IndexExpr(e),
+            Expression::Binary(e) => Node::BinaryExpr(e),
+            Expression::Unary(e) => Node::UnaryExpr(e),
+            Expression::PipeExpr(e) => Node::PipeExpr(e),
+            Expression::Call(e) => Node::CallExpr(e),
+            Expression::Conditional(e) => Node::ConditionalExpr(e),
+            Expression::StringExpr(e) => Node::StringExpr(e),
+            Expression::Paren(e) => Node::ParenExpr(e),
+            Expression::Integer(e) => Node::IntegerLit(e),
+            Expression::Float(e) => Node::FloatLit(e),
+            Expression::StringLit(e) => Node::StringLit(e),
+            Expression::Duration(e) => Node::DurationLit(e),
+            Expression::Uint(e) => Node::UintLit(e),
+            Expression::Boolean(e) => Node::BooleanLit(e),
+            Expression::DateTime(e) => Node::DateTimeLit(e),
+            Expression::Regexp(e) => Node::RegexpLit(e),
+            Expression::PipeLit(e) => Node::PipeLit(e),
```

Comment:
Seems like the other expression literals do not keep the `Lit` suffix. Should this be `Expression:Pipe`? or perhaps the other literals should be suffixed?

Reply:
This is an exception, there are two Pipe types one is `PipeLit` the other is `PipeExpr` so I couldn't use the bare `Pipe` constructor name.  So I used the full name in this case. 

Same for `StringLit` and `StringExpr`.